proto/buf/alpha/registry/v1alpha1/resolve.proto

```diff
@@ -107,3 +111,24 @@ message GetLocalModulePinsResponse {
   // This includes the transitive deps.
   repeated buf.alpha.module.v1alpha1.ModulePin dependencies = 2;
 }
+
+message ResolveReferenceRequest {
+  PluginRef plugin = 1;
+  ModuleRef module = 2;
+}
+
+message PluginRef {
+  string owner = 1;
+  string name = 2;
+  string version = 3;
+}
+
+message ModuleRef {
+  string owner = 1;
+  string repository = 2;
+  string reference = 3;
+}
```

Comment:
could/should we use [`LocalModuleReference`](https://github.com/bufbuild/buf/blob/206dd163499d3c287b43718597fbf7775d3b3283/proto/buf/alpha/registry/v1alpha1/module.proto#L21-L29) and [`CuratedPluginReference`](https://github.com/bufbuild/buf/blob/206dd163499d3c287b43718597fbf7775d3b3283/proto/buf/alpha/registry/v1alpha1/plugin_curation.proto#L212-L221) for these? not sure how we feel about using messages across services.

Otherwise, `s/Ref/Reference`

Reply:
Is there a reason why we would't need to send the `remote` for the module and plugin references?

Reply:
i guess you only want to resolve it for one bsr instance, which would be the instance that you're sending the request to?
ie. you _couldn't_ do this:
if you resolve the following:
```
buf beta registry resolve buf.build/bufbuild/eliza petstore.buf.build/petstore/petstore-generator
```

Reply:
We don't have any CLI commands that take multiple inputs, and the rest of our API surface does not take a remote for routing

Reply:
Yeah, that's valid on the inputs. I guess we would check that the plugin and module input are routing to the same remote, and then we wouldn't need to pass the `remote` onto the API.

Reply:
so `PluginReference` [already exists](https://github.com/bufbuild/buf/blob/fece8ff2cc6a184815a81551d94859384893e06c/proto/buf/alpha/registry/v1alpha1/generate.proto#L49) in this protobuf package, but it's got a field that I _don't_ want: `parameters`:

```
message PluginReference {
  // The owner of the plugin which identifies the
  // plugins to use with this generation.
  string owner = 1;
  // The name of the plugin which identifies the
  // plugins to use with this generation.
  string name = 2;
  // The plugin version to use with this generation.
  string version = 3;
  // The parameters to pass to the plugin. These will
  // be merged into a single, comma-separated string.
  repeated string parameters = 4;
}
```

I would use [LocalModuleReference](https://github.com/bufbuild/buf/blob/206dd163499d3c287b43718597fbf7775d3b3283/proto/buf/alpha/registry/v1alpha1/module.proto#L21-L29) but it's not referencing a local module? also [CuratedPluginReference](https://github.com/bufbuild/buf/blob/206dd163499d3c287b43718597fbf7775d3b3283/proto/buf/alpha/registry/v1alpha1/plugin_curation.proto#L212-L221) is every plugin that _can_ be uploaded to a BSR a "curated" plugin, because i assume this could refer to any plugin?

Reply:
I think it's "Local" in the sense that it's within the same remote as the plugin (can't have different remotes), and I _think_ it has to be a "curated" plugin to be a part of remote packages - not sure if the recent custom plugin changes make that different though.